tests/checkpointing/test_legacy_checkpoints.py

```diff
@@ -95,3 +97,33 @@ def test_resume_legacy_checkpoints(tmpdir, pl_version: str):
     # trainer.fit(model)
 
     sys.path = orig_sys_paths
+
+
+class OldStatefulCallback(Callback):
+    def __init__(self, state):
+        self.state = state
+
+    @property
+    def state_id(self):
+        return type(self)
+
+    def on_save_checkpoint(self, *args):
+        return {"state": self.state}
+
+    def on_load_checkpoint(self, trainer, pl_module, callback_state):
+        self.state = callback_state["state"]
+
+
+def test_resume_callback_state_saved_by_type(tmpdir):
```

Comment:
I'm not sure this is the correct file for this test.

Testing the legacy state id is not really related to what we call "legacy checkpoints"

Reply:
yes, I partially agree
better to move to tests/callbacks/test_callbacks.py?